### PR TITLE
bn256.Scalar to mod.Int

### PR DESCRIPTION
**Pull request by Daeinar**
This PR removes code duplication in bn256.Scalar by using the existing mod.Int library.

Merge sequence:
1. #292 
2. #291 
3. #290 
4. #289

### CODE REVIEW

File: pairing/bn256/suite_test.go [hunk 1]
```diff
@@ -27,25 +28,31 @@ func TestScalarMarshal(t *testing.T) {
...
+	require.True(t, d.Equal(e))
+	d.One()
+	e.One()
+	d.Mul(a, b)
+	d.Mul(d, c)
+	d.Inv(d)
+	e.Mul(a, b)
+	e.Mul(e, c)
+	e.Mul(e, d)
+	require.True(t, e.Equal(suite.G1().Scalar().One()))
```
Comment (kc1212) [inline pairing/bn256/suite_test.go]: For these kind of code I'd like to see an equation in the comments to explain what is happening.

Comment (Daeinar) [reply pairing/bn256/suite_test.go]: Done.

**Comment by ineiti** [comment]
Can you please rebase this on the latest kyber-merge? Else it's a mix of the previous PR and the new PR.

```
git checkout modint
git fetch
git rebase origin/master
git push --force
```

**Comment by Daeinar** [comment]
@ineiti: Done.

**Pull request merged**